File: Data Structures & Algorithms/palindromic-substrings/submission-3.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int ans =0;
        vector<vector<bool>>dp(s.size(), vector<bool>(s.size(),false));
        for(int i=s.size()-1;i>=0;--i){
            for(int j=i;j<s.size();++j){
                if(s[i]==s[j] and (j-i<=2 or dp[i+1][j-1]==true)){
                    dp[i][j]=true;
                    ++ans;
                }
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        ans, charSet = 0, set(s)

        for c in charSet:
            l =0
            charCount = 0
            for r in range(len(s)):
                if s[r]==c:
                    charCount +=1
                while r-l+1 - charCount >k:
                    if s[l]==c:
                        charCount -= 1
                    l += 1
                # print(c, l, r, charCount)
                ans = max(ans, r-l+1)
        return ans